### DISCUSSION TITLE
Version manifests?

### DISCUSSION THREAD

**Discussion by felipesere** [comtrya/comtrya#69] [Ideas]
At the moment all is evolving fast, so we could call this a "v0" manifest.
Once more and more people rely on it, I wonder if we should stabilise the manifest format (mostly which keys etc)
and then mark it as v1.

That way, if we ever decide to remove stuff we can have Comtrya check the version and warn users that certain manifests can't be read with the current binary?

Necessary? Gold plating?

**Comment by rawkode** [comment]
Really cool idea. Lots of thoughts pop in my head. 

Would we allow this to be set in the config or within a manifest itself? Can consumers of Comtrya have a repository with manifests or various versions?

I think versioning the manifest format is important, so I'm really happy you raised this.

**Comment by bketelsen** [comment]
I was just thinking the same thing.  and a `comtrya vet` command while we're at it.

**Comment by felipesere** [comment]
I think the manifests should state what version they support. That way, I can pull in dotfiles from a friends repo and be warned when it should not be able to apply

**Comment by rawkode** [comment]
This feels even more important now.

Last year, I added support for `Comtrya.yaml` to exist, which allows the manifest directory to be configured.

I don't suspect we'll be in a world where someone would mix manifest versions in their repository, but if we ever end up down the path of a central hub of shared manifests then this could likely happen.

I'm going to suggest we open an issue for this, allow `Comtrya.yaml` to set the default version. Then later, we can allow individual manifests to explicitly override this behaviour.

Thoughts?
